plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,166 @@
+package warp10
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+type Warp10 struct {
+	Prefix string
+
+	WarpUrl string
+
+	Token string
+}
+
+var sampleConfig = `
+  # prefix for metrics class Name
+  prefix = "telegraf."
+  ## POST HTTP Mode ##
+  # Url name of the Warp 10 server
+  warp_url = "localhost:4242/"
+  # Token to access your app on warp 10
+  token = "Token"
+`
+
+type MetricLine struct {
+	Metric    string
+	Timestamp int64
+	Value     string
+	Tags      string
+}
+
+func (o *Warp10) Connect() error {
+	return nil
+}
+
+func (o *Warp10) Write(metrics []telegraf.Metric) error {
+	if len(metrics) == 0 {
+		return nil
+	}
+	var timeNow = time.Now()
+	collectString := make([]string, 0)
+	index := 0
+	for _, mm := range metrics {
+
+		for k, v := range mm.Fields() {
+
+			metric := &MetricLine{
+				Metric:    fmt.Sprintf("%s%s", o.Prefix, mm.Name()+"."+k),
+				Timestamp: timeNow.Unix() * 1000000,
+			}
+
+			metricValue, buildError := buildValue(v)
+			if buildError != nil {
+				fmt.Printf("Warp: %s\n", buildError.Error())
+				continue
+			}
+			metric.Value = metricValue
+
+			tagsSlice := buildTags(mm.Tags())
+			metric.Tags = fmt.Sprint(strings.Join(tagsSlice, ","))
+
+			messageLine := fmt.Sprintf("%v// %s{%s} %v \n", metric.Timestamp, metric.Metric, metric.Tags, metric.Value)
+
+			collectString = append(collectString, messageLine)
+			index += 1
+		}
+	}
+	payload := fmt.Sprint(strings.Join(collectString, "\n"))
+	//defer connection.Close()
+	req, err := http.NewRequest("POST", o.WarpUrl, bytes.NewBufferString(payload))
+	req.Header.Set("X-Warp10-Token", o.Token)
+	req.Header.Set("Content-Type", "text/plain")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	fmt.Println("response Status:", resp.Status)
+	fmt.Println("response Headers:", resp.Header)
+	body, _ := ioutil.ReadAll(resp.Body)
+	fmt.Println("response Body:", string(body))
+
+	return nil
+}
+
+func buildTags(ptTags map[string]string) []string {
+	sizeTags := len(ptTags)
+	sizeTags += 1
+	tags := make([]string, sizeTags)
+	index := 0
+	for k, v := range ptTags {
+		tags[index] = fmt.Sprintf("%s=%s", k, v)
+		index += 1
+	}
+	tags[index] = fmt.Sprintf("source=telegraf")
+	sort.Strings(tags)
+	return tags
+}
+
+func buildValue(v interface{}) (string, error) {
+	var retv string
+	switch p := v.(type) {
```

Comment:
Hello thanks for this advice. As I am not so familiar with GO and our input format looks like the OpenTSDB one, I used their telegraf plugin as an inspiration, and that's why. I am going to change it then.